### ISSUE TITLE
Code Editor 2: GM crash when adding a new JSDoc header via Quick Fix if JSDoc is present already

### ISSUE THREAD

**Issue by Emc1923** [YoYoGames/GameMaker-Bugs#5275]
### Description

If you use quick fix (ctrl+Q) to create a jsdoc header when there's already one present, the IDE will go unstable and then it will crash to desktop.

The following error will populate in the log:

```
[11:57:19:73(da43)] Unhandled Exception: System.ArgumentOutOfRangeException
[11:57:19:74(da43)] Exception message: Index and count must refer to a location within the string. (Parameter 'count')
[11:57:19:104(da43)] Exception stack:    at System.String.Remove(Int32 startIndex, Int32 count)
   at YoYoStudio.TextEditor2.LspClient.ApplyWorkspaceEdit(WorkspaceEdit wsEdit)
   at YoYoStudio.TextEditor2.DocumentCodeAction.Apply()
   at YoYoStudio.GUI.Gadgets.TextEditor2.Components.CodeActionsComponent.<>c__DisplayClass18_0.<<OpenContextMenu>b__3>d.MoveNext()
--- End of stack trace from previous location ---
   at System.Threading.Tasks.Task.<>c.<ThrowAsync>b__128_1(Object state)
   at System.Threading.QueueUserWorkItemCallbackDefaultContext.Execute()
   at System.Threading.ThreadPoolWorkQueue.Dispatch()
   at System.Threading.PortableThreadPool.WorkerThread.WorkerThreadStart()
   at System.Threading.Thread.StartCallback()
[11:57:19:104(da43)] --------------------------------------------------------------------------------
[11:57:19:114(da43)] ####################################
[11:57:19:114(da43)] Assert in AddWindow (WindowManager:2419)
[11:57:19:115(da43)] Reason: You can't manipulate UI on a background thread
[11:57:19:115(da43)] ####################################
```
[ui.log](https://github.com/YoYoGames/GameMaker-Bugs/files/14851172/ui.log)
![jsdocheader](https://github.com/YoYoGames/GameMaker-Bugs/assets/105791928/5aacece2-618f-4c64-b90f-55be43c2aa8d)

### Steps To Reproduce

1. Start GameMaker
2. Make sure CE2 is enabled
3. Open a project and create a new script
4. Add a jsdoc header manually
5. Attempt to add a new jsdoc header using feather quickfix
6. See the issue

### How reliably can you recreate this issue using your steps above?

Always

### Which version of GameMaker are you reporting this issue for?

2024.400 (Betas)

### Which operating system(s) are you seeing the problem on?

Windows 11, Windows 10

### Are you running GameMaker from inside your Steam library?

No

### Contact Us Package Attached?

- [X] I have attached my Contact Us Package

### Sample Project Added?

- [ ] I have included a small sample project

**Comment by zreedy** [comment]
Additionally, the Undo stack is corrupted on if jsdoc is applied to an empty script and then you undo. The result is deleting the function signature!

**Comment by zreedy** [comment]
The primary ticket is now fixed, the linked tickets will be addressed shortly.
Additionally, the following undocumented bugs have been fixed:
 - Applying any Quick Fix will result in new lines being applied when undo/redoing the file
 - Closing a file after a Quick Fix has been applied results in the top 2 lines being joined despite not previously being joined
 - Applying a Quick Fix results in the Client and Server descyncing their documents

**Comment by Emc1923** [comment]
Verified fixed as of  IDE v2024.400.0.538  Runtime v2024.400.0.557.
